src/Analyzers.CodeFixes/CSharp/CodeFixes/UseExplicitlyOrImplicitlyTypedArrayCodeFixProvider.cs

```
@@ -39,29 +55,40 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+
+        context.RegisterCodeFix(codeAction, diagnostic);
+    }
+
+    public static async Task<Document> ApplyFixToDocumentAsync(Document document, Diagnostic diag, CancellationToken cancellationToken)
+    {
+        SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (!TryFindFirstAncestorOrSelf(
+            root,
+            diag.Location.SourceSpan,
+            out SyntaxNode node,
+            predicate: f => f.IsKind(SyntaxKind.ImplicitArrayCreationExpression, SyntaxKind.ArrayCreationExpression)))
         {
-            case ImplicitArrayCreationExpressionSyntax implicitArrayCreation:
-                {
-                    CodeAction codeAction = CodeAction.Create(
-                        "Use explicitly typed array",
-                        ct => ChangeArrayTypeToExplicitAsync(document, implicitArrayCreation, ct),
-                        GetEquivalenceKey(diagnostic));
-
-                    context.RegisterCodeFix(codeAction, diagnostic);
-                    break;
-                }
-            case ArrayCreationExpressionSyntax arrayCreation:
-                {
-                    CodeAction codeAction = CodeAction.Create(
-                        "Use implicitly typed array",
-                        ct => ChangeArrayTypeToImplicitAsync(document, arrayCreation, ct),
-                        GetEquivalenceKey(diagnostic));
-
-                    context.RegisterCodeFix(codeAction, diagnostic);
-                    break;
-                }
+            return null;
         }
+
+        return node switch
+        {
+            ImplicitArrayCreationExpressionSyntax implicitArrayCreation => await ChangeArrayTypeToExplicitAsync(document, implicitArrayCreation, cancellationToken).ConfigureAwait(false),
+            ArrayCreationExpressionSyntax arrayCreation => await ChangeArrayTypeToImplicitAsync(document, arrayCreation, cancellationToken).ConfigureAwait(false),
+            _ => null
```

This arm is unreachable, so I would prefer to throw InvalidOperationException.